modules/mapper-extras/src/main/java/org/opensearch/index/mapper/ScaledFloatFieldMapper.java

```diff
@@ -188,6 +189,14 @@ public ScaledFloatFieldType(String name, double scalingFactor) {
             this(name, true, false, true, Collections.emptyMap(), scalingFactor, null);
         }
 
+        @Override
+        public byte[] encodePoint(Number value) {
+            assert value instanceof Double;
+            byte[] point = new byte[Long.BYTES];
+            LongPoint.encodeDimension((long) (scalingFactor * value.doubleValue()), point, 0);
```

Comment:
Because scaled float is saved as long. I just did another look around in this class and found out that there are existing utility methods to reuse.
`LongPoint.encodeDimension(Math.round(scale(value)), point, 0)`